provider/google.go

```diff
@@ -267,16 +254,58 @@ func (p *googleProvider) submitChange(zone string, change *dns.Change) error {
 		return nil
 	}
 
-	_, err := p.changesClient.Create(p.project, zone, change).Do()
+	zones, err := p.Zones()
 	if err != nil {
-		if !isNotFound(err) {
-			return err
+		return err
+	}
+
+	// separate into per-zone change sets to be passed to the API.
+	changes := separateChange(zones, change)
+
+	for z, c := range changes {
+		if _, err := p.changesClient.Create(p.project, z, c).Do(); err != nil {
+			if !isNotFound(err) {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+// separateChange separates a multi-zone change into a single change per zone.
+func separateChange(zones map[string]*dns.ManagedZone, change *dns.Change) map[string]*dns.Change {
+	changes := make(map[string]*dns.Change)
+
+	for _, z := range zones {
+		changes[z.Name] = &dns.Change{
+			Additions: []*dns.ResourceRecordSet{},
+			Deletions: []*dns.ResourceRecordSet{},
+		}
+
+		for _, a := range change.Additions {
+			if strings.HasSuffix(ensureTrailingDot(a.Name), z.DnsName) {
+				changes[z.Name].Additions = append(changes[z.Name].Additions, a)
```

Comment:
doesn't this lead to record being created in multiple zones, if its suffix matches multiple zones?